sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/RestrictionTracker.java

```diff
@@ -99,6 +100,10 @@
    */
   public abstract void checkDone() throws IllegalStateException;
 
+  public boolean isBounded() throws NotImplementedException {
```

Comment:
We should use the return type `IsBounded`.

Reply:
Are you suggesting that using `PCollection.IsBounded` as return type? Any reason that a boolean doesn't work here?

Reply:
Your right in using PCollection.IsBounded would be weird.

The purpose of not using a boolean on PCollection was to make it clear whether something was bounded or unbounded. Boolean values also require the person to remember what the contract of the method is.

We can define an enum in RestrictionTracker similar to what we do in PCollection.

Reply:
I would prefer having Enum in RestrictionTracker. `isBounded` should mean current restriction is bounded or not, kind of different from what `PCollection.IsBounded` means.

Reply:
sgtm